### PR TITLE
Add support for protected TLVs

### PR DESCRIPTION
**Pull request by utzig**
Protected TLVs is a new feature supported by MCUBoot beginning with version 1.4.0. This allows the hash+signature to also include a few TLVs that should be protected from modification by users. For more details:

https://github.com/JuulLabs-OSS/mcuboot/blob/master/docs/design.md#protected-tlvs

### CODE REVIEW

**Comment by utzig** [comment]
@ccollins476ad This was tested on HW already, but I think there are a few things pending for a complete PR, like those `FindTlvXXX` functions that are are used by manufacturing images...

/cc @agross-korg I would suggesting adding your encryption TLVs as protected TLVs once this PR is merged!

File: image/image.go [hunk 1]
```diff
@@ -287,6 +296,19 @@ func (i *Image) RemoveTlvsWithType(tlvType uint8) []ImageTlv {
 	})
 }
 
+// ImageTrailer constructs an image trailer corresponding to the given image.
```
Comment (ccollins476ad) [inline image/image.go]: This comment seems a bit inaccurate.

Comment (utzig) [reply image/image.go]: Fixed, thanks!

**Comment by utzig** [comment]
@agross-korg Please take a look if you are OK with this, I just added your encryption related TLVs as protected TLVs, which means that on your calls to `bootutil_tlv_iter_begin` the last parameter must be `true`.

**Comment by agross-korg** [comment]
lgtm.  I tested this and it works.

**Comment by utzig** [comment]
Btw, this was tagged `v0.0.8` and `newt` can be updated later.

**Pull request merged**